### DISCUSSION TITLE
JSX followed by markdown in the same line does not work: forgets switching back to `md` mode after `react`

### DISCUSSION THREAD

**Discussion by Domiii** [mdx-js/mdx#1876] [Q&A] [answered]
Input: `<span>hi</span> [a](b)`

* Expected:
  > <span>hi</span> [a](b)
* Actual: 
  > hi &lbrack;a](b)

**Q: Are there any workarounds?**

## Notes

* It works fine the other way around, i.e.: `[a](b) <span>hi</span>`
* Switching back to `md` inline works fine in GitHub Markdown. E.g.: <span>hi</span> [a](b)
* Moving the `md` part to a second line does not work either. It needs to be completely separated to reset.
* It's fixed in `@mdx-js/react` :/

## System

Windows 10

`@mdx-js/react@1.6.22`

```
> yarn why @mdx-js/react
yarn why v1.22.4
[1/4] Why do we have the module "@mdx-js/react"...?
[2/4] Initialising dependency graph...
[3/4] Finding dependency...
[4/4] Calculating file sizes...
=> Found "@mdx-js/react@1.6.22"
info Has been hoisted to "@mdx-js/react"
info Reasons this module exists
   - Specified in "dependencies"
   - Hoisted from "@docusaurus#core#@docusaurus#mdx-loader#@mdx-js#react"
   - Hoisted from "@docusaurus#preset-classic#@docusaurus#theme-classic#@mdx-js#react"
   - Hoisted from "@docusaurus#core#@docusaurus#utils#@mdx-js#runtime#@mdx-js#react"
```

**Comment by Domiii** [comment, marked as answer]
I found a workaround (with [@josh-cena](https://github.com/Josh-Cena)'s help):

* Josh pointed out that if the line DOES NOT START with React, it works fine.
* -> Putting an [invisible space](https://www.google.com/search?q=invisible+space+html&hl=en) in front fixes it.

 `&#8203;<span>hi</span> [a](b)` works as expected.

**Reply by MingweiSamuel** [reply]
This doesn't actually work, this just makes it treat the "JSX" as inline markdown _HTML_

**Comment by wooorm** [comment]
- It’s good to get the terms right. React is not JSX and JSX is not React. Here you mean JSX.
- Use the latest (next) version of MDX. See the website and docs for more info: https://mdxjs.com.
- In particular, see the “What is MDX?” (specifically, “interleaving”) article: https://mdxjs.com/docs/what-is-mdx/

**Reply by Domiii** [reply]
Thanks for the response.

* I shall improve my verbiage accordingly!
* Luckily, I was able to find a hackfix/workaround for `v1`
* Upgrading to `v2` is out of my hands (I'm using docusaurus, not `@mdx-js` directly).
